<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="Ela" failures="28" tests="187" name="Default test" time="155.79" errors="0" timestamp="2025-03-27T16:32:28 CDT">
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="3.275"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.583"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.978"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.648">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [404] but found [401]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [404] but found [401]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.LoginandLogout_Step.the_admin_get_invalid_endpoint_response_code_and_message_as_and(LoginandLogout_Step.java:104)
at ✽.The Admin get invalid Endpoint response code and message as "login" and "TC_04_login"(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/01Login.feature:8)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.822"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.918"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.64"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.78"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.828"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.601"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.73"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.721"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.815">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [201] but found [400]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [201] but found [400]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Program_Step.the_admin_get_missing_additional_field_response_code_and_message_as_and_for_program(Program_Step.java:167)
at ✽.The Admin get missingAdditionalField response code and message as "program" and "TC_07_post_program" for Program(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/02Program.feature:11)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.809"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.814">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [200] but found [400]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [200] but found [400]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Program_Step.the_admin_get_valid_update_program_byprogram_id_response_code_and_message_as_and_for_update_program(Program_Step.java:485)
at ✽.The Admin get valid update program byprogramId response code and message as "program" and "TC_01_put_program" for update Program(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/02Program.feature:28)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.517"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.606">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [400] but found [401]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [400] but found [401]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Program_Step.the_admin_get_update_by_invalid_postbody_response_code_and_message_as_and_for_update_program(Program_Step.java:523)
at ✽.The Admin get update by invalid Postbody response code and message as "program" and "TC_03_put_program" for update Program(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/02Program.feature:28)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.817">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [400] but found [401]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [400] but found [401]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Program_Step.the_admin_get_update_by_without_payload_response_code_and_message_as_and_for_update_program(Program_Step.java:540)
at ✽.The Admin get update by without payload response code and message as "program" and "TC_04_put_program" for update Program(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/02Program.feature:28)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.918"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.815"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.715"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.721"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.61"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.507"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.432"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.341"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.522"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.912"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.055"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.364"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.525"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.527"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.639"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.365"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.47"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.475"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.402"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.486">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [400] but found [201]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [400] but found [201]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Batch_Step.the_admin_get_batch_existingbatchname_response_code_and_message_as_and(Batch_Step.java:68)
at ✽.The Admin get Batch  existingbatchname response code and message as "batch" and "TC_02_batch"(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/03Batch.feature:11)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.36"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.316"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.353"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.292"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.524"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="6.515"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.551"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.934"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.328"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.322"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.322"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.525"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.461"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.356"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.332"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.481"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.32"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.508">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [404] but found [200]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [404] but found [200]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Batch_Step.the_admin_get_batchby_program_id_retrievethebatchidbyinactiveprogramid_response_code_and_message_as_and(Batch_Step.java:356)
at ✽.The Admin get BatchbyProgramId  retrievethebatchidbyinactiveprogramid response code and message as "batch" and "TC_20_batch"(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/03Batch.feature:65)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.446"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.321"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.311"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.324"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.297"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.341"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.321"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.307"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.48"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.485"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.45"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.365"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.415"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.732">
    <failure type="java.lang.AssertionError" message="Mismatch in userComments expected [null] but found [nnhhh]">
      <![CDATA[java.lang.AssertionError: Mismatch in userComments expected [null] but found [nnhhh]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at requestBuilder.UserRequest.validateDataUserPost(UserRequest.java:658)
at stepDefinitions.User_Step.the_admin_get_valid_data_response_code_and_message_as_and_for_user(User_Step.java:45)
at ✽.The Admin get valid data response code and message as "user" and "TC_01_user" for User(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/04User.feature:11)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.321">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [201] but found [400]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [201] but found [400]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.User_Step.the_admin_get_mandatory_fields_response_code_and_message_as_and_for_user(User_Step.java:57)
at ✽.The Admin get mandatory fields response code and message as "user" and "TC_02_user" for User(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/04User.feature:11)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.597"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.607"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.82"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="3.009"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="2.256"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.658"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.912"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.511"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="5.12"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.51"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.511"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.512"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.922"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="2.936"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.963">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [200] but found [400]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [200] but found [400]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.User_Step.the_admin_get_update_user_by_userid_valid_response_code_and_message_as_and_for_user(User_Step.java:273)
at ✽.The Admin get update user by userid valid response code and message as "user" and "TC_17_user" for User(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/04User.feature:45)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.832"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.618"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.921">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [200] but found [400]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [200] but found [400]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.User_Step.the_admin_get_update_userrole_status_valid_response_code_and_message_as_and_for_user(User_Step.java:310)
at ✽.The Admin get update userrole status valid response code and message as "user" and "TC_20_user" for User(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/04User.feature:45)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.029">
    <failure type="java.lang.ClassCastException" message="Cannot cast java.lang.String to java.lang.Integer">
      <![CDATA[java.lang.ClassCastException: Cannot cast java.lang.String to java.lang.Integer
at java.base/java.lang.Class.cast(Class.java:4139)
at common.Utils.get(Utils.java:37)
at requestBuilder.UserRequest.userPutPgmBatchRequestBody(UserRequest.java:160)
at requestBuilder.UserRequest.updateUserByPgmBatch(UserRequest.java:552)
at stepDefinitions.User_Step.the_admin_sends_https_get_request_for_assign_user_to_pgm_batch_valid_scenarios_as_input_and_for_user(User_Step.java:316)
at ✽.The Admin sends HTTPS Put request for assign user to pgm batch valid scenarios as input "user" and "TC_21_user" for User(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/04User.feature:44)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.028">
    <failure type="java.lang.ClassCastException" message="Cannot cast java.lang.String to java.lang.Integer">
      <![CDATA[java.lang.ClassCastException: Cannot cast java.lang.String to java.lang.Integer
at java.base/java.lang.Class.cast(Class.java:4139)
at common.Utils.get(Utils.java:37)
at requestBuilder.UserRequest.userPutPgmBatchRequestBody(UserRequest.java:160)
at requestBuilder.UserRequest.updateUserByPgmBatch(UserRequest.java:552)
at stepDefinitions.User_Step.the_admin_sends_https_get_request_for_assign_user_to_pgm_batch_invalid_scenarios_as_input_and_for_user(User_Step.java:330)
at ✽.The Admin sends HTTPS Put request for assign user to pgm batch invalid scenarios as input "user" and "TC_22_user" for User(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/04User.feature:44)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.028">
    <failure type="java.lang.ClassCastException" message="Cannot cast java.lang.String to java.lang.Integer">
      <![CDATA[java.lang.ClassCastException: Cannot cast java.lang.String to java.lang.Integer
at java.base/java.lang.Class.cast(Class.java:4139)
at common.Utils.get(Utils.java:37)
at requestBuilder.UserRequest.userPutPgmBatchRequestBody(UserRequest.java:160)
at requestBuilder.UserRequest.updateUserByPgmBatch(UserRequest.java:552)
at stepDefinitions.User_Step.the_admin_sends_https_put_request_for_assign_user_to_pgm_batch_missing_required_field_scenarios_as_input_and_for_user(User_Step.java:343)
at ✽.The Admin sends HTTPS Put request for assign user to pgm batch missing required field scenarios as input "user" and "TC_23_user" for User(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/04User.feature:44)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.723">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [200] but found [400]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [200] but found [400]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.User_Step.the_admin_get_update_user_login_status_valid_response_code_and_message_as_and_for_user(User_Step.java:394)
at ✽.The Admin get update user login valid response code and message as "user" and "TC_24_user" for User(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/04User.feature:45)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.126"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.812"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.739"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.716">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [400] but found [201]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [400] but found [201]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Class_Step.the_admin_get_invalid_data_response_code_and_message_as_and_for_class(Class_Step.java:105)
at ✽.The Admin get invalid data response code and message as "class" and "TC_04_class_post" for Class(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/05Class.feature:11)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.827">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [400] but found [201]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [400] but found [201]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Class_Step.the_admin_get_existing_class_topic_response_code_and_message_as_and_for_class(Class_Step.java:124)
at ✽.The Admin get existing class topic response code and message as "class" and "TC_05_class_post" for Class(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/05Class.feature:11)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.094"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.928"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.682"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.846"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.615">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [400] but found [401]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [400] but found [401]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Class_Step.the_admin_get_additional_fields_response_code_and_message_as_and_for_class(Class_Step.java:87)
at ✽.The Admin get additional fields response code and message as "class" and "TC_03_class_put" for Class(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/05Class.feature:28)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.511"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.611">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [400] but found [200]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [400] but found [200]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Class_Step.the_admin_get_invalid_topic_response_code_and_message_as_and_for_class(Class_Step.java:223)
at ✽.The Admin get invalid Topic response code and message as "class" and "TC_05_class_put" for Class(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/05Class.feature:28)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.713"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.612"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.617"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.711"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.469"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.86">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [400] but found [200]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [400] but found [200]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Class_Step.the_admin_get_invalidData_allrecordings_response_code_and_message_as_and_for_class(Class_Step.java:303)
at ✽.The Admin get invalidData allrecordings response code and message as "class" and "TC_11_classAllRecord_put" for Class(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/05Class.feature:28)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.514"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.612"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.903"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.662"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.945"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.885"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.623"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.027">
    <failure type="java.lang.IllegalArgumentException" message="parameterValue cannot be null">
      <![CDATA[java.lang.IllegalArgumentException: parameterValue cannot be null
at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:501)
at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:485)
at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)
at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:60)
at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:86)
at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:57)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
at io.restassured.internal.common.assertion.AssertParameter.notNull(AssertParameter.groovy:26)
at io.restassured.internal.common.assertion.AssertParameter$notNull.callStatic(Unknown Source)
at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:53)
at io.restassured.internal.common.assertion.AssertParameter$notNull.callStatic(Unknown Source)
at io.restassured.internal.RequestSpecificationImpl.pathParam(RequestSpecificationImpl.groovy:515)
at requestBuilder.ClassRequest.classGetByIdAndTopic(ClassRequest.java:364)
at stepDefinitions.Class_Step.the_admin_sends_https_get_request_for_get_class_by_staff_id_valid_scenarios_as_input_and_for_class(Class_Step.java:440)
at ✽.The Admin sends HTTPS Get request for Get class by staffId valid scenarios as input "class" and "TC_14_class_get" for Class(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/05Class.feature:50)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.899"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.714"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.614"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.917"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.614"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.613"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.507"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.512"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.819"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.613"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.921"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.016"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.719"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.72"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.614"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.018">
    <failure type="java.lang.IllegalArgumentException" message="parameterValue cannot be null">
      <![CDATA[java.lang.IllegalArgumentException: parameterValue cannot be null
at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:501)
at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:485)
at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)
at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:60)
at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:86)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
at io.restassured.internal.common.assertion.AssertParameter.notNull(AssertParameter.groovy:26)
at io.restassured.internal.common.assertion.AssertParameter$notNull.callStatic(Unknown Source)
at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:53)
at io.restassured.internal.common.assertion.AssertParameter$notNull.callStatic(Unknown Source)
at io.restassured.internal.RequestSpecificationImpl.pathParam(RequestSpecificationImpl.groovy:515)
at requestBuilder.ClassRequest.classGetByIdAndTopic(ClassRequest.java:358)
at stepDefinitions.Class_Step.the_admin_sends_https_get_request_for_scenarios_as_input_and_for_class_invalid(Class_Step.java:502)
at ✽.The Admin sends HTTPS Get request for scenarios as input "class" and "TC_22_class_get" for Class invalid(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/05Class.feature:67)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.688"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.616"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.51"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.506"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.621"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.609"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.615"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.512"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.608"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.821"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.618"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.927"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.331"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.985"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.51"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.355"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.393"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.541"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.012">
    <failure type="java.lang.IllegalArgumentException" message="parameterValue cannot be null">
      <![CDATA[java.lang.IllegalArgumentException: parameterValue cannot be null
at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:501)
at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:485)
at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:73)
at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:60)
at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:86)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
at io.restassured.internal.common.assertion.AssertParameter.notNull(AssertParameter.groovy:26)
at io.restassured.internal.common.assertion.AssertParameter$notNull.callStatic(Unknown Source)
at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:53)
at io.restassured.internal.common.assertion.AssertParameter$notNull.callStatic(Unknown Source)
at io.restassured.internal.RequestSpecificationImpl.pathParam(RequestSpecificationImpl.groovy:515)
at requestBuilder.ClassRequest.classGetByIdAndTopic(ClassRequest.java:364)
at stepDefinitions.Class_Step.the_admin_sends_class_get_allclasses_bystaffid_request_with_no_auth_as_input_and_for_class(Class_Step.java:612)
at ✽.The Admin sends class_get_allclasses_bystaffid request with no auth as input "class" and "TC_06_class_noAuth" for class(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/06NoauthandInvalidURL.feature:19)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.588"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.459"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.487"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.846"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.928"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.876"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.671"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.825"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.806"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.931"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.612"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.923"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.803"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.541"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.901"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.923"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.507"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.799"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.046"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.689"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.86">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [401] but found [404]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [401] but found [404]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Program_Step.the_admin_gets_response_code_and_message_as_and_for_program(Program_Step.java:686)
at ✽.The Admin gets response code and message as "program" and "TC_07_program_noAuth" for program(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/06NoauthandInvalidURL.feature:69)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.799">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [401] but found [200]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [401] but found [200]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Program_Step.the_admin_gets_response_code_and_message_as_and_for_program(Program_Step.java:686)
at ✽.The Admin gets response code and message as "program" and "TC_08_program_noAuth" for program(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/06NoauthandInvalidURL.feature:69)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.017">
    <failure type="java.lang.NullPointerException" message="Cannot invoke &amp;quot;String.trim()&amp;quot; because the return value of &amp;quot;java.util.Map.get(Object)&amp;quot; is null">
      <![CDATA[java.lang.NullPointerException: Cannot invoke "String.trim()" because the return value of "java.util.Map.get(Object)" is null
at requestBuilder.ProgramRequest.setEndpointProgram(ProgramRequest.java:65)
at requestBuilder.ProgramRequest.programPost(ProgramRequest.java:119)
at stepDefinitions.Program_Step.the_admin_sends_createclass_request_with_invalid_url_as_input_and_for_program(Program_Step.java:746)
at ✽.The Admin sends createclass request with invalid url as input "class" and "TC_01_program_InvalidURL" for program(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/06NoauthandInvalidURL.feature:85)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="2.555"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.608">
    <failure type="java.lang.AssertionError" message="The following asserts failed:
	Status code mismatch! expected [200] but found [404]">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Status code mismatch! expected [200] but found [404]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at requestBuilder.CommonRequest.validateStatusCode(CommonRequest.java:70)
at stepDefinitions.Program_Step.the_admin_get_response_code_and_message_as_and_for_delete_program(Program_Step.java:631)
at ✽.The Admin get response code and message as "program" and "TC_03_delete_program" for delete Program(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/07Cleanup.feature:20)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.013"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.848"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.793"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.558">
    <failure type="java.lang.NullPointerException" message="Cannot invoke &amp;quot;java.lang.CharSequence.toString()&amp;quot; because &amp;quot;replacement&amp;quot; is null">
      <![CDATA[java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.toString()" because "replacement" is null
at java.base/java.lang.String.replace(String.java:3177)
at requestBuilder.UserRequest.deleteUser(UserRequest.java:646)
at stepDefinitions.User_Step.the_admin_sends_https_delete_request_for_invalid_user_id_scenarios_as_input_and_for_class(User_Step.java:374)
at ✽.The Admin sends HTTPS Delete request for invalid userID scenarios as input "user" and "TC_02_user_del" for Class(file:///C:/Users/ems11/Desktop/ElamathiProjects/Team04_RestfulRebels_APIPhase2/Team04_RestfulRebels/src/test/resources/features/07Cleanup.feature:36)
]]>
    </failure>
  </testcase> <!-- runScenario -->
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="1.395"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="4.095"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.6"/>
  <testcase classname="io.cucumber.testng.AbstractTestNGCucumberTests" name="runScenario" time="0.518"/>
</testsuite> <!-- Default test -->
